tests/ipfwd/conftest.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+from ipaddress import ip_address
+import logging
+import json
+
+
+logger = logging.getLogger(__name__)
+
+
+'''
+In case of multi-dut we need src_host_ip, src_router_ip, dst_host_ip, src_ptf_port_list, dst_ptf_port_list for the dut under test, 
+to take care of that made changes in the testcase 
+'''
+
+def get_lag_facts(dut, lag_facts, switch_arptable, mg_facts, ignore_lags, test_facts, key='src'):
+    if not mg_facts['minigraph_portchannels']:
+        pytest.fail("minigraph_portchannels is not defined")
+
+    # minigraph facts
+    up_lag = None
+    for a_lag_name, a_lag_data in lag_facts['lags'].items():
+        if a_lag_data['po_intf_stat'] == 'Up' and a_lag_name not in ignore_lags:
+            # We found a portchannel that is up.
+            up_lag = a_lag_name
+            test_facts[key + '_port_ids'] = [mg_facts['minigraph_ptf_indices'][intf] for intf in a_lag_data['po_config']['ports']]
+            test_facts[key + '_router_mac'] = dut.facts['router_mac']
+            for intf in mg_facts['minigraph_portchannel_interfaces']:
+                if intf['attachto'] == up_lag:
+                    addr = ip_address(unicode(intf['addr']))
+                    if addr.version == 4:
+                        test_facts[key + '_router_ipv4'] = intf['addr']
+                        test_facts[key + '_host_ipv4'] = intf['peer_addr']
+                        test_facts[key + '_host_mac'] = switch_arptable['arptable']['v4'][intf['peer_addr']]['macaddress']
+                    elif addr.version == 6:
+                        test_facts[key + '_router_ipv6'] = intf['addr']
+                        test_facts[key + '_host_ipv6'] = intf['peer_addr']
```

Comment:
Can you explicitly return gathered facts instead of implicitly updating the `test_facts` dict argument  to return values? This kind of design could be confusing and error prone.

Reply:
Made the requested changes